Marlin/src/module/stepper.cpp

```diff
@@ -2094,7 +2094,10 @@ hal_timer_t Stepper::calc_timer_interval(uint32_t step_rate) {
 
   #ifdef CPU_32_BIT
 
-    return uint32_t(STEPPER_TIMER_RATE) / step_rate; // A fast processor can just do integer division
+    // A fast processor can just do integer division
+    if (step_rate) return uint32_t(STEPPER_TIMER_RATE) / step_rate;
```

Comment:
i.e., Always returns something <= `STEPPER_TIMER_RATE`. If we had a float incoming, it would be possible to get values > `STEPPER_TIMER_RATE` for `0 < step_rate < 1`. Do we consider an incoming `step_rate` of 0 to be a value between 0 and 1. If so, the return value could be `> STEPPER_TIMER_RATE` and `<= HAL_TIMER_TYPE_MAX`.